c2cgeoform/static/src/index.js

```diff
@@ -16,84 +16,96 @@ const widgets = []
 let itemIcon
 
 export function initMap(target, options) {
-  const { center, zoom, projection } = options.view
   const source = new VectorSource()
-  source.on('addfeature', () => map.getView().fit(source.getExtent()))
   let vectorLayer = createVectorLayer(source)
   const context = { feature: null }
   vectorLayer.setStyle(getStyleFunction({ opacity: 0.5, context }))
 
   let map = new Map({
-    layers: [createBaseLayer(options.baselayer), vectorLayer],
+    layers: options.baseLayers
+      .map(def => createBaseLayer(def))
+      .concat([vectorLayer]),
     target,
-    view: new View({ center, zoom, projection }),
+    view: new View(options.view || {}),
   })
+  if (options.view.initialExtent) {
+    map.getView().fit(options.view.initialExtent)
+  }
+
   if (options.url)
     fetch(options.url)
       .then(resp => resp.json())
       .then(json => format.readFeatures(json))
       .then(features => {
         source.addFeatures(features)
-        options.onFeatureLoad(features)
+        if (options.fitSource) {
+          map.getView().fit(source.getExtent(), {
+            maxZoom: options.fitMaxZoom || 18,
+          })
+        }
+        if (options.onFeaturesLoaded) {
+          options.onFeaturesLoaded(features)
+        }
       })
 
   // Change feature style on Hover
   map.on('pointermove', e => {
     if (e.dragging) return
+    if (context.feature) {
+      context.feature.__over = false
+    }
     let feature
     map.forEachFeatureAtPixel(e.pixel, f => (feature = f), { hitTolerance: 3 })
     map.getTargetElement().classList.toggle('hovering', !!feature)
     context.feature = feature
+    if (context.feature) {
+      context.feature.__over = true
```

Comment:
can't you just unset `context.feature` instead of storing status inside the feature?

Reply:
I changed this because I initially planned to allow creating the style function in the caller (where context is not accessible) and pass it in options.